src/backend/distributed/sql/udfs/citus_prepare_pg_upgrade/12.1-1.sql

```diff
@@ -84,6 +84,13 @@ BEGIN
        objects.colocationid
     FROM pg_catalog.pg_dist_object objects,
          pg_catalog.pg_identify_object_as_address(objects.classid, objects.objid, objects.objsubid) address;
+
+    -- if we are upgrading from PG14/PG15 to PG16+,
+    -- we will need to regenerate the partkeys because they will include varnullingrels as well.
+    -- so we save the partkeys as column names here
+    CREATE TABLE IF NOT EXISTS public.pg_dist_partkeys_pre_upgrade AS
```

Comment:
I'm still in favor of calling this pg_dist_partkeys_pre_upgrade_16, and checking if `varnullingrels` exists for the row.

I'm suggesting that because, say if you are upgrading from PG16 to PG17, we are needlessly modifying the rows.

is there any problem with that?